NitroxModel/DataStructures/Optional.cs

```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```

Comment:
You forgot to change the class definition to `public struct Optional<T> : ISerializable, IEquatable<Optional<T>> where T : class` so that the equality methods can be used by systems depending on the IEquatable-abstraction.